### ISSUE TITLE
Add `slideChange` event to `amp-scrollable-carousel`

### ISSUE THREAD

**Issue by cathyxz** [ampproject/amphtml#14505]
Follow up to https://github.com/ampproject/amphtml/pull/14477. 
Add `slideChange` event where currently makes sense. Basically, it's a bit ambiguous which slide index we are on when the user is manually scrolling. But we should be able to unambiguously fire this event whenever `goToSlide` is called.

**Comment by kmturley** [comment]
I need this event for a type carousel which needs to show indicator dots.

There is a Google code lab showing <amp-carousel> using the slideChange event to update indicator dots:
https://github.com/googlecodelabs/advanced-interactivity-in-amp/blob/master/static/final.html#L216

However when you change the `type="slides"` to `type="carousel"`, the event stops working. I believe because  it checks for a page deck:
https://github.com/ampproject/amphtml/blob/26d1a9107976857fdf029143a31291e51cdae5d7/extensions/amp-gwd-animation/0.1/amp-gwd-animation.js#L148

Is there a way to either have another type of event? or change the carousel buttons to trigger a custom function to set the carousel index?
```
<amp-carousel id="carouselWithPreview"></amp-carousel>
<amp-selector class="dots" id="carouselWithPreviewSelector" 
on="select:carouselWithPreview.goToSlide(index=event.targetOption)"></amp-selector>
```

**Comment by nainar** [comment]
This issue is fixed for all users who upgrade their `<amp-carousel>` version to 0.2. 

For more details please see documentation here: https://github.com/ampproject/amphtml/blob/master/extensions/amp-carousel/amp-carousel.md

**Comment by garapa** [comment]
It isn't fixed on version 0.2. Am I missing something?

https://github.com/ampproject/amphtml/blob/090da63a06a928cb92f98dff59775606c01c2b30/extensions/amp-carousel/0.2/amp-carousel.js#L679-L683
